src/web/app/pages-session/session-submission-page/session-submission-page.component.spec.ts

```diff
@@ -1,24 +1,572 @@
+    recipientList: [{ recipientName: 'Barry Harris', recipientIdentifier: 'barry-harris-id' }],
+    recipientSubmissionForms: [testRankOptionsRecipientSubmissionForm],
+    numberOfEntitiesToGiveFeedbackToSetting: NumberOfEntitiesToGiveFeedbackToSetting.UNLIMITED,
+    customNumberOfEntitiesToGiveFeedbackTo: 5,
+    showResponsesTo: [FeedbackVisibilityType.RECIPIENT, FeedbackVisibilityType.INSTRUCTORS],
+    showGiverNameTo: [FeedbackVisibilityType.RECIPIENT, FeedbackVisibilityType.INSTRUCTORS],
+    showRecipientNameTo: [FeedbackVisibilityType.RECIPIENT, FeedbackVisibilityType.INSTRUCTORS],
+  };
+
+  const testRankRecipientsQuestionSubmissionForm: QuestionSubmissionFormModel = {
+    feedbackQuestionId: 'feedback-question-id-rank-recipients',
+    questionNumber: 10,
+    questionBrief: 'question brief',
+    questionDescription: 'question description',
+    questionType: FeedbackQuestionType.RANK_RECIPIENTS,
+    questionDetails: {
+      minOptionsToBeRanked: 1,
+      maxOptionsToBeRanked: 2,
+      areDuplicatesAllowed: false,
+    } as FeedbackRankRecipientsQuestionDetails,
+    giverType: FeedbackParticipantType.INSTRUCTORS,
+    recipientType: FeedbackParticipantType.STUDENTS,
+    recipientList: [{ recipientName: 'Barry Harris', recipientIdentifier: 'barry-harris-id' }],
+    recipientSubmissionForms: [testRankRecipientsRecipientSubmissionForm],
+    numberOfEntitiesToGiveFeedbackToSetting: NumberOfEntitiesToGiveFeedbackToSetting.UNLIMITED,
+    customNumberOfEntitiesToGiveFeedbackTo: 5,
+    showResponsesTo: [FeedbackVisibilityType.RECIPIENT, FeedbackVisibilityType.INSTRUCTORS],
+    showGiverNameTo: [FeedbackVisibilityType.RECIPIENT, FeedbackVisibilityType.INSTRUCTORS],
+    showRecipientNameTo: [FeedbackVisibilityType.RECIPIENT, FeedbackVisibilityType.INSTRUCTORS],
+  };
+
+  const testInfo: AuthInfo = {
+    masquerade: false,
+    user: {
+      id: 'user-id',
+      isAdmin: false,
+      isInstructor: false,
+      isStudent: true,
+    },
+  };
+
+  const testQueryParams: any = {
+    courseid: 'CS3281',
+    fsname: 'Feedback Session Name',
+    key: 'reg-key',
+  };
+
+  const getFeedbackSessionArgs: any = {
+    courseId: testQueryParams.courseid,
+    feedbackSessionName: testQueryParams.fsname,
+    intent: Intent.STUDENT_SUBMISSION,
+    key: testQueryParams.key,
+    moderatedPerson: '',
+    previewAs: '',
+  };
+
   let component: SessionSubmissionPageComponent;
   let fixture: ComponentFixture<SessionSubmissionPageComponent>;
+  let authService: AuthService;
+  let navService: NavigationService;
+  let studentService: StudentService;
+  let instructorService: InstructorService;
+  let feedbackSessionsService: FeedbackSessionsService;
+  let feedbackResponsesService: FeedbackResponsesService;
+  let feedbackResponseCommentService: FeedbackResponseCommentService;
+  let feedbackQuestionsService: FeedbackQuestionsService;
+  let simpleModalService: SimpleModalService;
+  let ngbModal: NgbModal;
```

Comment:
should we use `const` instead of `let`

Reply:
The initialisation for these variables are done only later, so we use `let` here first